### PR TITLE
IGAPP-872: Add a hint about missing cities

### PR DESCRIPTION
**Pull request by SteffiStuffel**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

Icons are just placeholders until the correct ones are finished.

### CODE REVIEW

File: web/src/routes/CityNotCooperatingPage.tsx [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+  display: flex;
+  justify-content: center;
+`
+
+const Icon = styled.img`
+  width: calc(20px + 5vw);
+  height: calc(20px + 5vw);
+  flex-shrink: 0;
+  padding: 8px;
+  /* object-fit: contain; */
```
Comment (f1sh1918) [inline web/src/routes/CityNotCooperatingPage.tsx:45]: rm comment if not needed

File: web/src/constants/cityNotCooperatingTemplate.ts [hunk 2]
```diff
@@ -0,0 +1,2 @@
+export const template =
+  'Sehr geehrte Damen und Herren,\n\nbei der Suche nach [ORT] in der INTEGREAT-App habe ich leider keinen Treffer bekommen. Ich wohne seit  einiger Zeit in Ihrer Kommune und Ich finde es hilfreich für mich und andere, wenn Sie auch bei Integreat mitmachen.\nIn der App finde ich in verschiedenen Sprachen viele wichtige Informationen für meinen Alltag und über die Region. Ich finde es hilfreich für mich und andere, wenn Sie auch bei Integreat mitmachen.\n\nUnter der Homepage: https://integreat-app.de/ erfahren Sie mehr. \n\nHerzliche Grüße'
```
Comment (f1sh1918) [inline web/src/constants/cityNotCooperatingTemplate.ts:2]: is there a reason why we don't put this in translations?

File: web/src/components/BackNavigationHeader.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+import { useHistory } from 'react-router-dom'
+import styled from 'styled-components'
+
+import BackIcon from '../assets/back-icon.png'
+import HeaderContainer from './HeaderContainer'
+
+const BackButton = styled.img`
+  width: 20px;
+  height: 20px;
+  padding: 40px 0 10px 10px;
```
Comment (f1sh1918) [inline web/src/components/BackNavigationHeader.tsx:11]:
```suggestion
  padding: 40px 0 10px 10px;
  cursor: pointer;
```

File: web/src/constants/cityNotCooperatingTemplate.ts (hunk 2 above)
Comment (SteffiStuffel) [reply web/src/constants/cityNotCooperatingTemplate.ts:2]: yes it should always be in german, as the cities who get these text don't speak arabic or hungarian.

**Comment by f1sh1918** [comment]
Found this issues mobile on iphone 6/7/8 Plus

image

File: web/src/routes/CityNotCooperatingPage.tsx [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+const ListItem = styled.div`
+  display: flex;
+  align-items: center;
+`
+
+const StepNumber = styled.div`
+  float: left;
+  border-radius: 50%;
+  line-height: 2rem;
+  width: 2rem;
```
Comment (f1sh1918) [inline web/src/routes/CityNotCooperatingPage.tsx:56]:
```suggestion
  min-width: 2rem;
```

**Review by f1sh1918 — APPROVED** [review]

**Review by f1sh1918 — CHANGES_REQUESTED** [review]
pls resolve step icon issue on mobile.
nice work :)

File: web/src/components/__tests__/CityNotCooperatingFooter.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+describe('CityNotCooperatingFooter', () => {
+  it('should render text and button', () => {
+    const { getByText } = render(
+      <BrowserRouter>
+        <CityNotCooperatingFooter languageCode='de' />
+      </BrowserRouter>,
+      { wrapper: wrapWithTheme }
+    )
+    expect(getByText('cityNotCooperating')).toBeDefined()
+    expect(getByText('cityNotCooperatingButton')).toBeDefined()
```
Comment (Taggotty) [inline web/src/components/__tests__/CityNotCooperatingFooter.spec.tsx:17]: Perhaps you could also add a test which asserts that clicking on the button will direct you to the `cityNotCooperating` Page.

**Review by Taggotty — APPROVED** [review]
Release Notes? :)

File: web/src/routes/CityNotCooperatingPage.tsx [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+          <ListHeading>{t('whatToDo')}</ListHeading>
+          <ListItem>
+            <StepNumber>1</StepNumber>
+            <StepExplanation>{t('findOutMail')}</StepExplanation>
+          </ListItem>
+          <ListItem>
+            <StepNumber>2</StepNumber>
+            <StepExplanation>{t('sendText')}</StepExplanation>
+          </ListItem>
+          <Button onClick={() => copyToClipboard()}>{isCopied ? t('textCopied') : t('copyText')}</Button>
```
Comment (klinzo) [inline web/src/routes/CityNotCooperatingPage.tsx:115]:
```suggestion
          <Button onClick={copyToClipboard}>{isCopied ? t('textCopied') : t('copyText')}</Button>
```

File: web/src/routes/__tests__/CityNotCooperatingPage.spec.tsx [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+    expect(template).toBeDefined()
+  })
+
+  it('should handle button click correctly', async () => {
+    const { getByText, queryByText } = renderPage()
+    expect(queryByText('textCopied')).toBeNull()
+    const button = getByText('copyText')
+    fireEvent.click(button)
+    await waitFor(() => expect(getByText('textCopied')).toBeDefined())
+    expect(navigator.clipboard.writeText).toBeCalled()
```
Comment (klinzo) [inline web/src/routes/__tests__/CityNotCooperatingPage.spec.tsx:49]: Minor/not necessary:
Since `toBeCalled` is just an alias for `toHaveBeenCalled` I'd favor using `toHaveBeenCalled` for consistency to avoid confusion.

File: web/src/components/CityNotCooperatingFooter.tsx [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+import cityNotCooperatingIcon from '../assets/cityNotCooperating.svg'
+import { createPath } from '../routes'
+
+const FooterContainer = styled.div`
+  background-color: ${props => props.theme.colors.backgroundAccentColor};
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  padding: 20px;
+  border-bottom: 2px solid black;
```
Comment (klinzo) [inline web/src/components/CityNotCooperatingFooter.tsx:17]: I know this is in the designs as well but for me it looks like a glitch, I think dark grey/or no line at all would look better. But again, just personal preference.

However, you should use a color from our theme (there should be black).

**Review by klinzo — CHANGES_REQUESTED** [review]
This should be possible to disable this in the build config. It does not really make sense for malte/aschaffenburg and is a little bit confusing/using wrong designs.

File: translations/translations.json [hunk 9]
```diff
@@ -293,6 +293,17 @@
       "loading": "加载中。。。"
     }
   },
+  "cityNotCooperating": {
+    "de": {
+      "callToAction": "Schlag Integreat Deiner\nStadt vor!",
+      "explanation": "Das gemeinnützige Konzept der Integreat-App hat Dich überzeugt und Du glaubst, auch Deine Stadt, Dein Landkreis oder Deine Gemeinde könnte von der digitalen Lösung profitieren? Dann kannst Du hiermit ganz einfach aktiv werden.",
+      "whatToDo": "Das kannst Du tun:",
+      "findOutMail": "E-Mail Adresse Deiner Stadt herausfinden",
```
Comment (klinzo) [inline translations/translations.json:301]: I think `Region` is the naming we use here.
```suggestion
      "findOutMail": "E-Mail Adresse Deiner Region herausfinden",
```

**Comment by klinzo** [comment]
Really nice work! Nice design and implementation :)

**Comment by SteffiStuffel** [comment]
> Release Notes? :)

There already is a release note.

File: web/src/components/__tests__/CityNotCooperatingFooter.spec.tsx (hunk 5 above)
Comment (SteffiStuffel) [reply web/src/components/__tests__/CityNotCooperatingFooter.spec.tsx:17]: good idea, added it.

File: translations/translations.json [hunk 10]
```diff
@@ -293,6 +293,17 @@
       "loading": "加载中。。。"
     }
   },
+  "cityNotCooperating": {
+    "de": {
+      "callToAction": "Schlag Integreat Deiner\nStadt vor!",
```
[the remark this answers is not in the document]
Comment (SteffiStuffel) [reply translations/translations.json:298]: yes it is, it in the designs like this. Does not really make sense to be in the translations tough.

File: web/src/routes/LandingPage.tsx [hunk 11]
```diff
@@ -23,15 +23,16 @@ const LandingPage = ({ cities, match }: PropsType): ReactElement => {
...
   const metaDescription = t('metaDescription', { appName: buildConfig().appName })
 
+  useScrollToTopOnMount()
+
   return (
     <Layout
       footer={
         <>
-          <CityNotCooperatingFooter languageCode={languageCode} />
+          {buildConfig().appName.startsWith('Integreat') && <CityNotCooperatingFooter languageCode={languageCode} />}
```
Comment (klinzo) [inline web/src/routes/LandingPage.tsx:32]: There should be a flag for this in the build config

**Comment by klinzo** [comment]
There are three hidden comments not yet addressed :)

File: web/src/components/BackNavigationHeader.tsx [hunk 12]
```diff
@@ -0,0 +1,24 @@
+import React, { ReactElement } from 'react'
```
[the remark this answers is not in the document]
Comment (SteffiStuffel) [reply web/src/components/BackNavigationHeader.tsx:1]: yeah maybe it would make sense to discuss this again, but lets open an other ticket for it, as this ticket was `Ready` and groomed properly and is now done. So there is not less effort if doing it later and we can avoid having this ticket open for long time and therefore less conflicts to fix.

[the remark this answers is not in the document]
Comment (klinzo) [reply web/src/components/BackNavigationHeader.tsx:1]: I think we can do a small vote on mattermost

File: build-configs/BuildConfigType.ts [hunk 13]
```diff
@@ -30,6 +30,8 @@ export type FeatureFlagsType = FixedCityType & {
   sentry: boolean
   // Enables error tracking to sentry, can be disabled by users.
   developerFriendly: boolean // Enables additional debugging output for devs (i18n, redux, hidden cities, version).
+  // Enables the option for users to suggest Integreat to their city
+  cityNotCooperating: boolean
```
Comment (klinzo) [inline build-configs/BuildConfigType.ts:34]: Reading this variable name I'd have no clue what it is about. I'd rename to something like `recommendToCityIfNotCooperating` or similar.

File: translations/translations.json [hunk 14]
```diff
@@ -2784,7 +2795,9 @@
       "timeout": "Standortbestimmung fehlgeschlagen",
       "refresh": "Liste der Orte aktualisieren",
       "pageTitle": "Ortsauswahl",
-      "metaDescription": "{{appName}} ist Ihr digitaler Guide für Deutschland. Finden Sie lokale Informationen, Veranstaltungen und Beratung. Immer aktuell und in Ihrer Sprache."
+      "metaDescription": "{{appName}} ist Ihr digitaler Guide für Deutschland. Finden Sie lokale Informationen, Veranstaltungen und Beratung. Immer aktuell und in Ihrer Sprache.",
+      "cityNotFound": "Deine Stadt nicht gefunden?",
```
Comment (klinzo) [inline translations/translations.json:2799]:
```suggestion
      "cityNotFound": "Deine Region nicht gefunden?",
```

File: web/src/routes/LandingPage.tsx [hunk 15]
```diff
@@ -23,13 +22,13 @@ const LandingPage = ({ cities, match }: PropsType): ReactElement => {
   const pageTitle = t('pageTitle')
   const metaDescription = t('metaDescription', { appName: buildConfig().appName })
 
-  useScrollToTopOnMount()
```
Comment (klinzo) [inline web/src/routes/LandingPage.tsx]: Why did you remove this?

Comment (SteffiStuffel) [reply web/src/routes/LandingPage.tsx]: Its no longer needed on Landing page, as navigating back with the browser automatically jumps to the top.

**Review by klinzo — APPROVED** [review]
Well done, I really like it :tada: We may have to adjust the texts/translations later on but for now this should be fine. If you want you can apply the suggestions made in mattermost but that is definitely optional.

**Review by Taggotty — APPROVED** [review]
Really like it! Well done! 👍🏼

**Review by f1sh1918 — APPROVED** [review]
nice work 👍
